### PR TITLE
First cut at including Thorntail

### PR DESCRIPTION
**Pull request by pilhuhn**
Please especially @kenfinnigan have a look

### CODE REVIEW

**Review by kenfinnigan — COMMENTED** [review]
My only thought is whether it's better to use `project-defaults.yml` to define the context web root instead of `jboss-web.xml`?

**Comment by pilhuhn** [comment]
@kenfinnigan You know TT so much better than I do. I found this as a solution to set the context root, so I used it.
How/were would I use the project-defaults.yml? Is that a TTv2 thingy?

**Comment by kenfinnigan** [comment]
so `project-defaults.yml` can be added to src/main/resources with content like:

```
swarm:
  context:
    path: '/mp-servicemesh-sample/serviceA'
```

That should do what you need, and it works in v2

File: pom.xml [hunk 1]
```diff
@@ -230,5 +231,117 @@
...
+              <groupId>org.wildfly.swarm</groupId>
+              <artifactId>config-api-runtime</artifactId>
+            </exclusion>
+          </exclusions>
+        </dependency>
+<!-- the next ones should be in the above, but unfortunately are missing in TT 2.0
+    See THORN-2093
+-->
+        <dependency>
+             <groupId>io.thorntail</groupId>
```
Comment (Emily-Jiang) [inline pom.xml]: Is this MP Open API jar? The dependency on MP 1.3 should do it.

Comment (pilhuhn) [reply pom.xml]: I hope this whole "additional things" section will go away soon :-)

**Comment by pilhuhn** [comment]
@kenfinnigan The "downside" of project-defaults.yml in `src/main/resources/`is that this would also be copied into a liberty (or other server) deployment, while the current approach has the file in `src/main/profiles/thorntail/` and is thus more 'isolated'.

**Comment by kenfinnigan** [comment]
Sorry, I wasn't necessarily meaning the file had to be in `src/main/resources` but that it needs to be processed like it is.

Having it in a custom Thorntail folder that gets added to the /target directory on build is fine.

**Review by kenfinnigan — APPROVED** [review]

**Review by jonhawkes — APPROVED** [review]

**Pull request merged**